vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -272,25 +272,34 @@ func (r RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver)
 	filters.Add("status", "running")
 	filters.Add("status", "paused")
 	filters.Add("status", "restarting")
-	containers, err := c.ContainerList(context.Background(), types.ContainerListOptions{
+
+	ctx, cancel := context.WithTimeout(context.Background(), dockerConnTimeout*time.Second)
+	defer cancel()
+	containers, err := c.ContainerList(ctx, types.ContainerListOptions{
 		All:    true,
 		Filter: filters,
 	})
 	if err != nil {
+		log.Errorf("ContainerList failed (err: %v)", err)
 		return err
 	}
 
 	log.Debugf("Found %d running or paused containers", len(containers))
 	for _, ct := range containers {
-		containerJSONInfo, err := c.ContainerInspect(context.Background(), ct.ID)
+		ctx_inspect, cancel_inspect := context.WithTimeout(context.Background(), dockerConnTimeout*time.Second)
+		defer cancel_inspect()
```

Comment:
Does the cancel_inspect var given its re-assigned in each iteration work ok to remove each created context? Would just the last context created be removed? Since its a loop should the cancel func be called explicitly vs. reassigning the var.